__tests__/app.test.js

```diff
@@ -160,27 +161,110 @@ describe('GET /api/articles/:article_id/comments', () => {
       .get('/api/articles/99999/comments')
       .expect(404)
       .then(({ body }) => {
-        expect(body.msg).toBe(`Article not found`);
+        expect(body.msg).toBe(`Not Found`);
       });
   });
-  test('200: Responds with an empty array if article_id exists, but has no comments', () => {
+});
+
+describe('POST /api/articles/:article_id/comments', () => {
+  test('201: Responds with a new comment created at a given article_id', () => {
+    const testComment = {
+      username: 'butter_bridge',
+      body: "when are y'all gonna start posting recipes on this site?",
+    };
+
     return request(app)
-      .get('/api/articles/2/comments')
-      .expect(200)
-      .then(({ body: { comments } }) => {
-        expect(comments.length).toBe(0);
-        expect(Array.isArray(comments)).toBe(true);
+      .post('/api/articles/1/comments')
+      .send(testComment)
+      .expect(201)
+      .then(({ body: { comment } }) => {
+        expect(comment).toMatchObject({
+          article_id: 1,
+          author: 'butter_bridge',
+          body: "when are y'all gonna start posting recipes on this site?",
+          comment_id: expect.any(Number),
+          created_at: expect.any(String),
+          votes: 0,
+        });
+      });
+  });
+  test('400: Responds with an error message if article_id is NaN', () => {
+    const testComment = {
+      username: 'lurker',
+      body: "i don't normally do this sort of thing",
+    };
+
+    return request(app)
+      .post('/api/articles/scrambled-egg/comments')
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Bad Request');
+      });
+  });
+  test('400: Responds with an error message if username does not exist in users table', () => {
+    const testComment = {
+      username: 'broth-baby',
+      body: 'i like soup',
+    };
+
+    return request(app)
+      .post('/api/articles/3/comments')
+      .send(testComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Bad Request');
+      });
+  });
+  test('404: Responds with an error message if username missing from request', () => {
+    const testComment = {
+      body: "post this anywhere I don't mind",
+    };
+
+    return request(app)
+      .post('/api/articles/4/comments')
+      .send(testComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Not Found');
+      });
+  });
+  test('404: Responds with an error message if body missing from request', () => {
+    const testComment = {
+      username: 'soVeryShy',
+    };
+
+    return request(app)
+      .post('/api/articles/5/comments')
+      .send(testComment)
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe('Not Found');
+      });
```

Comment:
Here we are sending objects that are missing keys from what's required, would we expect a 404 "Not Found" error?

Reply:
I second-guessed myself before pushing this :weary: I'll update it to 400!